### ISSUE TITLE
Inference results are different every time

### ISSUE THREAD

**Issue by z-lai** [Adapter-Hub/adapter-transformers#64]
Hi,

I built my own sentiment-analysis-adapter but I found the inference result are different although the same text was input.
It is necessary to freeze the layers during inference or is there anything wrong with training configuration?

Training code (similar with: https://colab.research.google.com/github/Adapter-Hub/website/blob/master/app/static/notebooks/Adapter_Quickstart_Training.ipynb#scrollTo=M6vjtq3NHtxS)

```
model_name = "cl-tohoku/bert-base-japanese-whole-word-masking"
# BERT model for Japanese
data_args = GlueDataTrainingArguments(task_name="sst-2", data_dir="./glue_data/yahoo_movie_reviews/")
# yahoo_movie_reviews is a dataset very similar with sst-2 but in Japanese
training_args = TrainingArguments(
    logging_steps=1000, 
    per_device_train_batch_size=32, 
    per_device_eval_batch_size=64, 
    save_steps=1000,
    evaluate_during_training=True,
    output_dir="./models/yahoo_movie_reviews",
    overwrite_output_dir=True,
    do_train=True,
    do_eval=True,
    do_predict=True,
    learning_rate=0.0001,
    num_train_epochs=10,
)
set_seed(training_args.seed)
num_labels = glue_tasks_num_labels[data_args.task_name]

tokenizer = AutoTokenizer.from_pretrained(model_name)
model = AutoModelWithHeads.from_pretrained(model_name)
model.add_adapter("sst-2", AdapterType.text_task)
model.train_adapter(["sst-2"])
model.add_classification_head("sst-2", num_labels=num_labels)
model.set_active_adapters([["sst-2"]])

train_dataset = GlueDataset(data_args, tokenizer=tokenizer)
eval_dataset = GlueDataset(data_args, tokenizer=tokenizer, mode="dev")

def compute_metrics(p: EvalPrediction):
    preds = np.argmax(p.predictions, axis=1)
    return glue_compute_metrics(data_args.task_name, preds, p.label_ids)

trainer = Trainer(
    model=model,
    args=training_args,
    train_dataset=train_dataset,
    eval_dataset=eval_dataset,
    compute_metrics=compute_metrics,
)

trainer.train()
trainer.evaluate()
```

Inference code is almost as same as https://colab.research.google.com/github/Adapter-Hub/website/blob/master/app/static/notebooks/Adapter_Quickstart_Inference.ipynb#scrollTo=2xwdA1sz7eZO

Thanks!
Lai

**Comment by calpt** [comment]
Hi @z-lai,

From what I can see, your training setup seems to look fine.

During inference, it is important to set the model to eval mode using PyTorch's `model.eval()` method. This deactivates components such as dropout layers. Otherwise, the outputs differ for every forward pass. You can also freeze the gradient updates of the model layers (using `freeze_model()` or `torch.no_grad()`), but that should not make a difference in the output values.

**Comment by z-lai** [comment]
Hi @calpt,

Thank you for your reply.
After I tuned my code following https://docs.adapterhub.ml/training.html instead of above one,
I find it works now:)
